Content.Server/GameObjects/Components/Cargo/CargoTelepadComponent.cs

```diff
@@ -0,0 +1,99 @@
+            receiver.OnPowerStateChanged += PowerUpdate;
+            
+        }
+
+        public override void OnRemove()
+        {
+            if (Owner.TryGetComponent(out PowerReceiverComponent? receiver))
+            {
+                receiver.OnPowerStateChanged -= PowerUpdate;
+            }
+            base.OnRemove();
+        }
+
+        public void QueueTeleport(CargoProductPrototype product)
+        {
+            _teleportQueue.Add(product);
+            TeleportLoop();
+        }
+
+        private void PowerUpdate(object? sender, PowerStateEventArgs args)
+        {
+            if (args.Powered && _currentState == CargoTelepadState.Unpowered) {
+                _currentState = CargoTelepadState.Idle;
+                if(Owner.TryGetComponent<SpriteComponent>(out var spriteComponent))
+                    spriteComponent.LayerSetState(0, "pad-idle");
+                TeleportLoop();
+            }
+            else if (!args.Powered)
+            {
+                _currentState = CargoTelepadState.Unpowered;
+                if (Owner.TryGetComponent<SpriteComponent>(out var spriteComponent))
+                    spriteComponent.LayerSetState(0, "pad-offline");
+            }
+        }
+        private void TeleportLoop()
+        {
+            if (_currentState == CargoTelepadState.Idle && _teleportQueue.Count > 0)
+            {
+                _currentState = CargoTelepadState.Charging;
+                if (Owner.TryGetComponent<SpriteComponent>(out var spriteComponent))
+                    spriteComponent.LayerSetState(0, "pad-idle");
+                Owner.SpawnTimer((int) (TeleportDelay * 1000), () =>
+                {
+                    if (!Deleted && !Owner.Deleted && _currentState == CargoTelepadState.Charging && _teleportQueue.Count > 0)
+                    {
+                        _currentState = CargoTelepadState.Teleporting;
+                        if (Owner.TryGetComponent<SpriteComponent>(out var spriteComponent))
+                            spriteComponent.LayerSetState(0, "pad-beam");
+                        Owner.SpawnTimer((int) (TeleportDuration * 1000), () =>
+                        {
+                            if (!Deleted && !Owner.Deleted && _currentState == CargoTelepadState.Teleporting && _teleportQueue.Count > 0)
+                            {
+                                EntitySystem.Get<AudioSystem>().PlayFromEntity("/Audio/machines/phasein.ogg", Owner, AudioParams.Default.WithVolume(-8f));
+                                Owner.EntityManager.SpawnEntity(_teleportQueue[0].Product, Owner.Transform.Coordinates);
+                                _teleportQueue.RemoveAt(0);
+                                if (Owner.TryGetComponent<SpriteComponent>(out var spriteComponent))
+                                    spriteComponent.LayerSetState(0, "pad-idle");
+                                _currentState = CargoTelepadState.Idle;
+                                TeleportLoop();
```

Comment:
Placeholder(TM) 
I just used no visualizers since I didn't want to bother with making anything except the server comp for something that will get deleted anyway